observability/src/main/java/io/grpc/observability/interceptors/LoggingChannelInterceptor.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.observability.interceptors;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.ExperimentalApi;
+import io.grpc.MethodDescriptor;
+
+/** A channel provider that injects logging interceptor. */
+@ExperimentalApi("https://github.com/grpc/grpc-java/issues/8869")
+public final class LoggingChannelInterceptor implements ClientInterceptor {
```

Comment:
Does it have to be public API?

Reply:
good catch. Removing it

Reply:
`public` can be removed.

Reply:
`public` cannot be removed since it is referenced from outside the package (the parent package)

Reply:
How many classes will be in `io.grpc.observability.interceptors` package? If no more than a couple of them, we can put them in `io.grpc.observability` then we won't have to make them look like a pubilc API.

Reply:
I agree the implementation can be large and involve a lot of classes. So will it look better to refactor it as following?

```java
@Internal
public final class InternalLoggingChannelInterceptor
```

Reply:
done